cli/genenotebook.js

```diff
@@ -356,35 +356,68 @@ addTranscriptome
         sampleName,
         replicaGroup,
         description,
-      }
+      },
     );
   })
   .exitOverride(customExitOverride(addTranscriptome));
 
-// add interproscan
+// Add interproscan file
 const addInterproscan = add.command('interproscan');
 
 addInterproscan
   .description('Add InterProScan results to a running GeneNoteBook server')
-  .usage('[options] <InterProScan gff3 output file>')
+  .usage('[options] <InterProScan gff3 or tsv output file>')
   .arguments('<file>')
-  .option('-u, --username <username>', 'GeneNoteBook admin username')
-  .option('-p, --password <password>', 'GeneNoteBook admin password')
+  .requiredOption(
+    '-u, --username <adminUsername>',
+    'GeneNoteBook admin username',
+  )
+  .requiredOption(
+    '-p, --password <adminPassword>',
+    'GeneNoteBook admin password',
+  )
   .option(
     '--port [port]',
-    'Port on which GeneNoteBook is running. Default: 3000'
+    'Port on which GeneNoteBook is running. Default: 3000',
   )
-  .action((file, { username, password, port = 3000 }) => {
+  .option(
+    '--format [parser]',
+    `Choose a parser for the interproscan output files. Parses .gff3 and .tsv
+    extensions.`,
+  )
+  .action((file, { username, password, port = 3000, parser }) => {
     if (typeof file !== 'string') addInterproscan.help();
-    const fileName = path.resolve(file);
 
+    const fileName = path.resolve(file);
     if (!(fileName && username && password)) {
       addInterproscan.help();
     }
-    new GeneNoteBookConnection({ username, password, port }).call(
-      'addInterproscan',
-      { fileName }
-    );
+
+    let parserType;
+    const extensionFile = path.extname(file).replace(/\./g, '');
+
+    if (['tsv', 'gff3'].includes(parser)) {
+      parserType = parser;
+    } else if (['tsv', 'gff3'].includes(extensionFile)) {
+      parserType = extensionFile;
+    } else {
+      logger.error('--format parameter is not defined');
```

Comment:
It seems like this is only an error when both `parser` is undefined and the file suffix is not `gff3` or `tsv`. The error message should probably reflect that, something like 'Must specify --format when file extension is not tsv or gff3'.

Reply:
No problem, I just fixed that. And add more precise indications when the error occurs.